### ISSUE TITLE
During querying provider Run line marker (class com.intellij.execution.lineMarker.RunLineMarkerProvider)

### ISSUE THREAD

**Issue by BorzdeG** [kotest/kotest#5555]
plugin: 3.3.0.4
IntelliJ IDEA 2019.1 Snapshot (Ultimate Edition)
Build #IU-191.5109.14, built on February 6, 2019
JRE: 1.8.0_202-release-1483-b4 amd64

```
During querying provider Run line marker (class com.intellij.execution.lineMarker.RunLineMarkerProvider)

java.lang.ArrayIndexOutOfBoundsException: 2
	at io.kotlintest.plugin.intellij.psi.UtilsKt.matchFunction2WithStringAndLambdaArgs(utils.kt:94)
	at io.kotlintest.plugin.intellij.psi.FunSpecStyle.testPath(FunSpecStyle.kt:14)
	at io.kotlintest.plugin.intellij.psi.FunSpecStyle.isTestElement(FunSpecStyle.kt:9)
	at io.kotlintest.plugin.intellij.KotlinTestRunLineMarkerContributor.getInfo(KotlinTestRunLineMarkerContributor.kt:22)
	at com.intellij.execution.lineMarker.RunLineMarkerProvider.getLineMarkerInfo(RunLineMarkerProvider.java:49)
	at com.intellij.codeInsight.daemon.impl.LineMarkersPass.queryProviders(LineMarkersPass.java:171)
	at com.intellij.codeInsight.daemon.impl.LineMarkersPass.lambda$doCollectInformation$3(LineMarkersPass.java:96)
	at com.intellij.codeInsight.daemon.impl.Divider.divideInsideAndOutsideInOneRoot(Divider.java:79)
	at com.intellij.codeInsight.daemon.impl.LineMarkersPass.doCollectInformation(LineMarkersPass.java:91)
	at com.intellij.codeHighlighting.TextEditorHighlightingPass.collectInformation(TextEditorHighlightingPass.java:69)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$null$1(PassExecutorService.java:430)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1166)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$2(PassExecutorService.java:423)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:586)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:532)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:86)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.doRun(PassExecutorService.java:422)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$run$0(PassExecutorService.java:398)
	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:164)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:218)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:396)
	at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:174)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
```

**Comment by sksamuel** [comment]
Which version of kotlintest ?

**Comment by BorzdeG** [comment]
`io.kotlintest`: `3.2.1`
`org.jetbrains.kotlin`: `1.3.21`
Project JDK: `oracle 8u201`

**Comment by sksamuel** [comment]
I'm not sure if this is causing your error, but you need to be on 3.3.0-snapshot to use the plugin.
https://oss.sonatype.org/content/repositories/snapshots/io/kotlintest/

**Comment by sksamuel** [comment]
But I think you error is just a parse issue. Can you show me the test that fails (I don't need what's inside the test method, just the "structure").

**Comment by BorzdeG** [comment]
init state:
```kotlin
import io.kotlintest.specs.FunSpec

class DemoTest : FunSpec({
  test("demo tt")
                         })
```

The error appears at the time of entering the opening brace of the test

**Comment by sksamuel** [comment]
Ahh I see. You've started the lambda arg but not completed it, thus it's not available as a PsiElement. Ok I'll fix that.

**Comment by sksamuel** [comment]
Thanks for the feedback, btw, please feel free to keep testing the plugin, it's not "officially" released yet.
With 3.3.0.4 and 3.3.0-snapshot you should be able to run tests which are fully formed.

**Comment by BorzdeG** [comment]
Not necessary. You can not run the lambda - just click with the mouse after the closing parenthesis of the test so that the error reappears.

**Comment by sksamuel** [comment]
Ok, well should be fairly easily for me to reproduce after work and fix it.

**Comment by sksamuel** [comment]
This is fixed in plugin release 3.3.0.5
Please try that once it's approved by Jetbrains.

**Comment by sksamuel** [comment]
3.3.0.5 has now been approved.
@BorzdeG @bbaldino

**Comment by BorzdeG** [comment]
@sksamuel The error persisted. But now reproduced as follows:

Scenario 1:
1) Select all the code in the file
2) delete it

Scenario 2:
1) Create an empty file
2) insert the test code into it

In both scenarios, the code is analogous to the one above.
Video record?

**Comment by sksamuel** [comment]
Yes that would be great thanks.

**Comment by BorzdeG** [comment]
http://www.mediafire.com/file/f660d4y302rxpj3/simplescreenrecorder-2019-02-14_12.43.43.ogg/file

**Comment by sksamuel** [comment]
Fixed in 3.3.0.6.
